internal/config/canary_config.go

```diff
@@ -160,10 +173,24 @@ func (c CanaryConfig) String() string {
 		TLSClientKey = "[Client key]"
 	}
 
+	// is one of SASL mechanisms needing user/password is enabled, using placeholders for them
+	SASLUser := ""
+	SASLPassword := ""
+	if c.SASLMechanism == sarama.SASLTypePlaintext {
+		if c.SASLUser != "" {
+			SASLUser = "[SASL user]"
+		}
+
+		if c.SASLPassword != "" {
+			SASLPassword = "[SASL password]"
+		}
+	}
```

Comment:
If we just supporting SASL/PLAIN shouldn't we have an `else` here to error on other mechanisms? I can see you handle this in `auth.go`, but it seems cleaner to do it here, no?

Reply:
this function is just for printing the configuration passed to the canary, it's kind of override of `toString` in Java :-) ... the if condition I am putting there is just for hiding the username/password with placeholders but nothing more.

Reply:
Ah, that's what I get for just scanning the diff. Thanks for the clarification.